pkg/controller/azure/compute/aks.go

```diff
@@ -0,0 +1,298 @@
+	clusterNamePrefix = "aks-"
+
+	errorClusterClient = "Failed to create cluster client"
+	errorCreateCluster = "Failed to create new cluster"
+	errorSyncCluster   = "Failed to sync cluster state"
+	errorDeleteCluster = "Failed to delete cluster"
+)
+
+var (
+	ctx           = context.Background()
+	result        = reconcile.Result{}
+	resultRequeue = reconcile.Result{Requeue: true}
+)
+
+// Add creates a new Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// Reconciler reconciles a Provider object
+type Reconciler struct {
+	client.Client
+	scheme     *runtime.Scheme
+	kubeclient kubernetes.Interface
+	recorder   record.EventRecorder
+
+	connect func(*azurecomputev1alpha1.AKSCluster) (aks.Client, error)
+	create  func(*azurecomputev1alpha1.AKSCluster, aks.Client) (reconcile.Result, error)
+	delete  func(*azurecomputev1alpha1.AKSCluster, aks.Client) (reconcile.Result, error)
+	sync    func(*azurecomputev1alpha1.AKSCluster, aks.Client) (reconcile.Result, error)
+	secret  func(*azurecomputev1alpha1.AKSCluster, aks.Client) (*corev1.Secret, error)
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	r := &Reconciler{
+		Client:     mgr.GetClient(),
+		scheme:     mgr.GetScheme(),
+		kubeclient: kubernetes.NewForConfigOrDie(mgr.GetConfig()),
+		recorder:   mgr.GetRecorder(controllerName),
+	}
+	r.connect = r._connect
+	r.create = r._create
+	r.sync = r._sync
+	r.delete = r._delete
+	r.secret = r._secret
+	return r
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to Provider
+	err = c.Watch(&source.Kind{Type: &azurecomputev1alpha1.AKSCluster{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Reconcile reads that state of the cluster for a Provider object and makes changes based on the state read
+// and what is in the Provider.Spec
+func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.Printf("reconciling %s: %v", azurecomputev1alpha1.AKSClusterKindAPIVersion, request)
+	// Fetch the Provider instance
+	instance := &azurecomputev1alpha1.AKSCluster{}
+	err := r.Get(ctx, request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Object not found, return.  Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// Create GKE Client
```

Comment:
copy/paste: this is not getting a GKE Client, it's getting an AKS client.  Both comment and the variable name should be changed.